### ISSUE TITLE
Implementation v2ray-plugin into iOS App

### ISSUE THREAD

**Issue by vadimux** [shadowsocks/v2ray-plugin#200]
Hi everyone, 
I've implemented Shadowsocks protocol into my app with Swift and everything works great. However I have additional task to connect our app to Shadowsocks Server with **v2ray-plugin**. As option I should add v2ray-plugin into my iOS app.
Unfortunately, I didn't find any information at all about such implementation (yes, I've bought Shadowrocket), only for MacOS there is. 
Could someone share information (links, instructions and etc) on how to correctly add support v2ray-plugin to the app for iOS?
Thanks and have a good day!

**Comment by bd901029** [comment]
How could you integrate v2ray-plugin on iOS? Are you interested if I ask to work with me? Thanks.

**Comment by vadimux** [comment]
> How could you integrate v2ray-plugin on iOS? Are you interested if I ask to work with me? Thanks.

as I wrote, I don't know how to add v2ray-plugin on iOS and my it was my question, because it's easy to add for Android and there are no any details for iOS

**Comment by bd901029** [comment]
Oh really sorry to hear that. Have you checked Mac version? Would it be many different with iOS?

**Comment by vadimux** [comment]
> Oh really sorry to hear that. Have you checked Mac version? Would it be many different with iOS?

yes, I've checked version for MacOS and there are global differences due to the specifics of operating systems

**Comment by bd901029** [comment]
I reviewed the MacOS client. Seems like it has not v2ray module, but system proxy configuration module only. Would it be the same with iOS? I think NEVPNManager or NEPacketTunnelProvider would be a solution?

**Comment by atlant04** [comment]
also looking for a solution, specifically targeting NEPacketTunnelProvider. The main concern is compiling v2ray plugin as a framework (gomobile maybe) and somehow integrating it with NEPacketTunnelProvider. Unfortunately, no solution just yet. Please, if someone manages to find something, text in this thread

**Comment by parsanoori** [comment]
@bd901029 Iranians are unable to pay for you application. There's no online payment method here.  
We are under an extreme censorships ever seen during our history these days.
Shadowsocks with v2ray-plugin is one of the very few options which we have to connect to global internet.  
Your application is the only one able to connect to shadowsocks using v2ray-plugin on iPhone.
Please and please consider a week of free installation.      
Thank you for your generous in advance.

**Comment by mahdimoqadasi** [comment]
Any update on this? Does anyone succeed on implement ios clients? Or how could we use Shell in iOS? I have a Mac version that uses v2ray-core, and it's working, but in iOS there is no "Process" class available to run a shell command. Can anyone help me through this? (if there is an ios source of v2ray client for sell, contact me)

**Comment by hamedpooramiri** [comment]
Any update on this?  no information on how to implement v2ray on iOS !!??

**Comment by loveCodingStar** [comment]
https://github.com/shadowsocks/v2ray-plugin/issues/200

**Comment by loveCodingStar** [comment]
https://t.me/JohnGG123

**Comment by netrolite** [comment]
we need this
